app/controllers/api/v1/mixins/rbac_mixin.rb

```diff
@@ -2,22 +2,30 @@ module Api
   module V1
     module Mixins
       module RBACMixin
-        VALID_RESOURCE_VERBS = %w[read write order].freeze
+        VALID_RESOURCE_VERBS = %w[create read update delete order].freeze
+        ADMINISTRATOR_ROLE_NAME = 'Catalog Administrator'.freeze
 
-        def write_access_check
-          resource_check('write')
+        def update_access_check
+          resource_check('update')
         end
 
         def read_access_check
           resource_check('read')
         end
 
+        def create_access_check
+          permission_check('create')
+        end
+
+        def delete_access_check
+          permission_check('create')
```

Comment:
Do we want 'delete' here instead of 'create'?

Reply:
Yes and I also need to add a spec